src/test/java/core/basesyntax/HelloWorldTest.java

```
@@ -1,8 +1,116 @@
  */
 public class HelloWorldTest {
 
+    private static RegistrationServiceImpl registerService;
+    private static StorageDao storageDao;
+    private User user;
+
+    @BeforeAll
+    static void setUp() {
+        registerService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+    }
+
+    @BeforeEach
+    void connect() {
+        user = new User();
+    }
+
+    @Test
+    void register_correctUser_ok() {
+        user.setLogin("Igor123");
+        user.setPassword("123456");
+        user.setAge(19);
+
+        registerService.register(user);
+        User newUser = storageDao.get(user.getLogin());
+
+        assertEquals(newUser.getLogin(), user.getLogin());
+    }
+
+    @Test
+    void register_nullAge_NotOk() {
+        user.setLogin("Igor123");
+        user.setPassword("123456");
+        user.setAge(null);
+
+        assertThrows(NullPointerException.class, () -> registerService.register(user));
+    }
+
+    @Test
+    void register_nullLogin_NotOk() {
+        user.setLogin(null);
+        user.setPassword("123456");
+        user.setAge(19);
+
+        assertThrows(NullPointerException.class, () -> registerService.register(user));
+    }
+
+    @Test
+    void register_nullPassword_NotOk() {
+        user.setLogin("Igor123");
+        user.setPassword(null);
+        user.setAge(19);
+
+        assertThrows(NullPointerException.class, () -> registerService.register(user));
+    }
+
+    @Test
+    void register_sizePassword_notOk() {
+        user.setLogin("Igor123");
+        user.setPassword("1234");
+        user.setAge(19);
+
+        User actual = registerService.register(user);
+        assertNotEquals(actual, user);
+    }
+
+    @Test
+    void register_sizeLogin_notOk() {
+        user.setLogin("Igor");
+        user.setPassword("123456");
+        user.setAge(19);
+
+        User actual = registerService.register(user);
+        assertNotEquals(actual, user);
```

Issue: The test expects a different user object to be returned for invalid login length, but the correct behavior should be to throw an exception for invalid input. Please update the test to expect an exception.